File: python/sanitizer.py
def sanitize_content(content: str) -> str:
    dangerous_patterns = [
        "IGNORE ALL PREVIOUS INSTRUCTIONS",
        "IGNORE PREVIOUS INSTRUCTIONS",
        "You are now",
        "Tu es maintenant",
        "Forget everything",
        "Oublie tout",
    ]

    for pattern in dangerous_patterns:
        if pattern.lower() in content.lower():
            print(f"⚠️  Prompt injection détecté: {pattern}")
            return "[CONTENU FILTRÉ: tentative d'injection détectée]"

    return content

File: java/src/main/java/com/datahub/Sanitizer.java
package com.datahub;

public class Sanitizer {
    
    public static String sanitizeContent(String content) {
        String[] dangerousPatterns = {
            "IGNORE ALL PREVIOUS INSTRUCTIONS",
            "IGNORE PREVIOUS INSTRUCTIONS",
            "You are now",
            "Tu es maintenant",
            "Forget everything",
            "Oublie tout"
        };

        String contentLower = content.toLowerCase();
        for (String pattern : dangerousPatterns) {
            if (contentLower.contains(pattern.toLowerCase())) {
                System.out.println("⚠️  Prompt injection détecté: " + pattern);
                return "[CONTENU FILTRÉ: tentative d'injection détectée]";
            }
        }

        return content;
    }
}
